profiles/schema.py

```diff
@@ -688,6 +693,102 @@ def mutate_and_get_payload(cls, root, info, **input):
         return CreateProfileMutation(profile=profile)
 
 
+class VerifiedPersonalInformationAddressInput(graphene.InputObjectType):
+    street_name = graphene.String()
+    postal_code = graphene.String()
+    city = graphene.String()
+
+
+class VerifiedPersonalInformationInput(graphene.InputObjectType):
+    first_name = graphene.String(description="First name(s).")
+    last_name = graphene.String(description="Last name.")
+    given_name = graphene.String(description="The name the person is called with.")
+    national_identification_number = graphene.String(
+        description="Can be social security number or other identifier."
+    )
+    email = graphene.String()
+    municipality_of_residence = graphene.String(
+        description="Official municipality of residence in Finland as a free form text."
+    )
+    municipality_of_residence_number = graphene.String(
+        description="Official municipality of residence in Finland as an official number."
+    )
+    permanent_address = graphene.InputField(
+        VerifiedPersonalInformationAddressInput,
+        description="The permanent residency address in Finland.",
+    )
+    temporary_address = graphene.InputField(
+        VerifiedPersonalInformationAddressInput,
+        description="The temporary residency address in Finland.",
+    )
+
+
+class ProfileWithVerifiedPersonalInformationInput(graphene.InputObjectType):
+    user_id = graphene.UUID(
+        required=True,
+        description="The **user id** of the user the Profile is or will be associated with.",
+    )
+    verified_personal_information = graphene.InputField(
+        VerifiedPersonalInformationInput, required=True
+    )
+
+
+class ProfileWithVerifiedPersonalInformationNode(graphene.ObjectType):
+    class Meta:
+        interfaces = (relay.Node,)
+
+
+class CreateOrUpdateProfileWithVerifiedPersonalInformationMutationPayload(
+    graphene.ObjectType
+):
+    profile = graphene.Field(ProfileWithVerifiedPersonalInformationNode)
+
+
+class CreateOrUpdateProfileWithVerifiedPersonalInformationMutation(graphene.Mutation):
+    class Arguments:
+        input = ProfileWithVerifiedPersonalInformationInput(required=True)
+
+    Output = CreateOrUpdateProfileWithVerifiedPersonalInformationMutationPayload
+
+    @staticmethod
+    def mutate(parent, info, input):
+        address_types = [{"name": "permanent_address"}, {"name": "temporary_address"}]
+
+        user_id_input = input.pop("user_id")
+        verified_personal_information_input = input.pop("verified_personal_information")
+
+        for address_type in address_types:
+            address_type["input"] = verified_personal_information_input.pop(
+                address_type["name"], None
+            )
+
+        (user, created) = User.objects.get_or_create(uuid=user_id_input)
```

Comment:
I don't think the parentheses are needed.

Reply:
Removed